pallas-crypto/build.rs

```diff
@@ -0,0 +1,92 @@
+use std::process::Command;
+
+macro_rules! ok (($expression:expr) => ($expression.unwrap()));
+macro_rules! log {
+    ($fmt:expr) => (println!(concat!("pallas-crypto/build.rs:{}: ", $fmt), line!()));
+    ($fmt:expr, $($arg:tt)*) => (println!(concat!("pallas-crypto/build.rs:{}: ", $fmt),
+    line!(), $($arg)*));
+}
+
+fn main() {
+    // Build and link libsodium
+    run("git", |command| {
+        command
+            .arg("submodule")
+            .arg("update")
+            .arg("--init")
+            .arg("--recursive")
+            .arg("--force")
+    });
+
+    // if windows
+    #[cfg(target_os = "windows")]
+    {
+        // Build libsodium with MSBuild
+        run("msbuild", |command| {
+            command
+                .current_dir("contrib/libsodium/builds/msvc/vs2019")
+                .arg("libsodium.sln")
+                .arg("/p:Configuration=StaticRelease")
+                .arg("/p:Platform=x64")
```

Comment:
does that mean it will fail when building on arm windows machines ?

Reply:
and it looks like we are also losing WASM support here too ?

Reply:
As far as I can gather, the vrf crate from IOG might not be compatible with libsodium. It seems to be waiting on a PR going into the curve25519_dalek crate that has been open for 2 years without being merged. https://github.com/dalek-cryptography/curve25519-dalek/pull/377

Maybe it is compatible though since it's pointing to the fork of that library. Not sure if that will mess up our dependency graph or not.  Anyway... off to do some testing.

Reply:
It's meant to be compatible with cardano just like the fork of libsodium that cardano use. it's forking the curve25519 crate and I don't think it should impact the dependencies (or have a build impact as big as depending on a C library).

Don't think it's particularly need any maintenance, which explains the 3years with no update.